src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -192,25 +194,32 @@ public String  MakeGUID() {
     public void BeginConnectPeer(String toPeerId, ConnectStatusCallback connectStatusCallback) {
         ConnectStatusCallback tmpCallback = connectStatusCallback;
 
+        // needed since mBtToRequestSocket can be disconnected from other thread
+        BtToRequestSocket tmp = mBtToRequestSocket;
+
         if(tmpCallback == null) {
             //nothing we should do, since we can not update progress
             print_debug("BeginConnectPeer callback is NULL !!!!!!");
-        }else if(mBtToRequestSocket != null) {
-            tmpCallback.ConnectionStatusUpdate("Already connected to " + mBtToRequestSocket.GetPeerId(),-1);
+        }else if(tmp != null) {
+            tmpCallback.ConnectionStatusUpdate("Already connected to " + tmp.GetPeerId(),-1);
```

Comment:
@DrJukka This assumes that we can connect to only one peer at a time. But I thought you said that Bluetooth supports multiple simultaneous connections to different peers?